lightning/src/util/test_utils.rs

```diff
@@ -124,16 +127,43 @@ impl<'a> chain::Watch for TestChainMonitor<'a> {
 
 		let ret = self.update_ret.lock().unwrap().clone();
 		if let Some(next_ret) = self.next_update_ret.lock().unwrap().take() {
-			*self.update_ret.lock().unwrap() = next_ret;
+			*self.update_ret.lock().unwrap() = Some(next_ret);
 		}
-		ret
+		if ret.is_some() {
+			return ret.clone().unwrap();
+		}
+		update_res
```

Comment:
This could use some commentary on why the `TestChainMonitor` result is preferred over the (possible) `TestPersister` result. Is it because `ChainMonitor` will favor an error from `ChannelMonitor::update_monitor` over an error from its persister?

Perhaps the adding the assert mentioned in my earlier comment would be sufficient.

Reply:
Added the `assert` :)

Reply:
Should there be an assert in this function, too? :)

Reply:
:woman_facepalming: ty lol